metricbeat/mb/module/wrapper.go

```diff
@@ -308,6 +308,9 @@ func (r reporterV2) Event(event mb.Event) bool {
 	if event.Took == 0 && !r.start.IsZero() {
 		event.Took = time.Since(r.start)
 	}
+	if r.msw.Module().Config().Period > 0 {
```

Comment:
This all LGTM, except that `Period` will be set to 10s even for the push metricsets (I think). So events from push metricsets will get a `period` and I'd like for that to not happen.

Can you check if this is the case? And if so, see if there is something else we can check to avoid adding `period` to push metricset events by default.

Reply:
I tried it with the http/server metricset and unfortunately you are right. Need to figure out how we can detect this.

Reply:
It should be possible from to determine from the `msw` value whether or not it's push or pull.

Maybe you could add an enum to the metricSetWrapper that identifies the type as being push vs pull. You'd want to do this rather than doing a type assertion on each event.